pennylane_lightning/core/src/simulators/lightning_tensor/tncuda/gates/tests/Test_MPSTNCuda_Param.cpp

```diff
@@ -0,0 +1,1177 @@
+// Copyright 2024 Xanadu Quantum Technologies Inc.
+
+// Licensed under the Apache License, Version 2.0 (the License);
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+
+// http://www.apache.org/licenses/LICENSE-2.0
+
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an AS IS BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <complex>
+#include <vector>
+
+#include <catch2/catch.hpp>
+
+#include "DevTag.hpp"
+#include "Gates.hpp"
+#include "MPSTNCuda.hpp"
+#include "TNCudaGateCache.hpp"
+#include "TestHelpers.hpp"
+
+/// @cond DEV
+namespace {
+using namespace Pennylane::LightningGPU;
+using namespace Pennylane::LightningTensor;
+using namespace Pennylane::LightningTensor::TNCuda::Gates;
+using namespace Pennylane::Util;
+using namespace Pennylane;
+namespace cuUtil = Pennylane::LightningGPU::Util;
+} // namespace
+/// @endcond
+
+TEMPLATE_TEST_CASE("MPSTNCuda::Gates::PhaseShift", "[MPSTNCuda_Param]", float,
```

Comment:
One way to make tests more concise might be to:
- make sure `applyMatrix` works as intended;
- tests named operations against `applyMatrix` as done [here](https://github.com/PennyLaneAI/pennylane-lightning/blob/master/pennylane_lightning/core/src/simulators/lightning_kokkos/gates/tests/Test_StateVectorKokkos_Param.cpp#L71). 
No need to change it now, but if maintaining the tests becomes an issue we could consider changing our approach.

Reply:
Will update it when adding the exact TN